NeuroID/src/advancedDeviceReactNativeLib/java/com/neuroid/tracker/extensions/AdvancedDeviceExtension.kt

```diff
@@ -0,0 +1,43 @@
+package com.neuroid.tracker.extensions
+
+import com.neuroid.tracker.NeuroID
+import com.fingerprintjs.android.fpjs_pro.Configuration
+import com.fingerprintjs.android.fpjs_pro.FingerprintJSFactory
+import com.neuroid.tracker.service.NIDAdvKeyService
+import com.neuroid.tracker.service.OnKeyCallback
+import com.neuroid.tracker.utils.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import com.neuroid.tracker.utils.FPJSHelper
+
+fun NeuroID.start(advancedDeviceSignals: Boolean) {
+    start()
+
+    if (advancedDeviceSignals) {
+        CoroutineScope(Dispatchers.IO).launch {
+            val keyService = NIDAdvKeyService()
+            var fpjsRetryCount = 0
+            val FPJS_RETRY_MAX = 3
+            keyService.getKey(object : OnKeyCallback {
+                override fun onKeyGotten(key: String) {
+                    val applicationContext = getApplicationContext()
+                    val fpjsClient = applicationContext?.let {
+                        FingerprintJSFactory(applicationContext = it).createInstance(
+                            Configuration(
+                                apiKey = key
+                            )
+                        )
+                    }
+                    //  Retrieving the Request ID from FPJS
+                    var fpjsHelper = FPJSHelper()
+                    fpjsHelper.getRequestId(fpjsClient, fpjsRetryCount, FPJS_RETRY_MAX)
```

Comment:
Should this just return the ID and then we log it here? The function name makes it sound like it will return something

Reply:
how about generatingRequestId?

Reply:
we can change the name to processRequestID() since this will not return a request id.

Reply:
changed to createRequestIdEvent